src/Microsoft.PowerShell.CoreCLR.AssemblyLoadContext/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -267,6 +281,97 @@ private Assembly Resolve(AssemblyLoadContext loadContext, AssemblyName assemblyN
             return asmLoaded;
         }
 
+        // Try to find the assembly in GAC by looking up the directories in well know locations.
+        // First try to find in GAC_MSIL, then depending on process bitness; GAC_64 or GAC32.
+        // If there are multiple version of the assembly, load the latest.
+        private bool TryFindInGAC(AssemblyName assemblyName, out string assemblyFilePath)
+        {
+            bool assemblyFound = false;
+            assemblyFilePath = null;
+            char dirSeparator = IO.Path.DirectorySeparatorChar;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && PowerShellAssemblyLoadContextTestHooks.AllowGACLoading)
```

Comment:
Wouldn't it be better to `#ifdef !UNIX` line 265 (calling this method) above as well as this whole method?